From 832fb9a26af053e989e59e5bae0aabbc7d2943a7 Mon Sep 17 00:00:00 2001
From: Ludwig Krispenz <lkrispen@redhat.com>
Date: Mon, 6 Oct 2014 17:12:42 +0200
Subject: [PATCH 16/21] Ticket 47918 - result of dna_dn_is_shared_config is
 incorrectly used

Bug Description:  dna_load_shared_servers() is called for all entries,
                  except the config entries

Fix Description:   revert the logic of the result of dna_dn_is_shared_config()

https://fedorahosted.org/389/ticket/47918

Reviewed by: Mark, Thanks

(cherry picked from commit 8d9679194c945a6e0b06f6527fba6b4548d4844d)
---
 ldap/servers/plugins/dna/dna.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/plugins/dna/dna.c b/ldap/servers/plugins/dna/dna.c
index b42150a..bc280a4 100644
--- a/ldap/servers/plugins/dna/dna.c
+++ b/ldap/servers/plugins/dna/dna.c
@@ -4163,7 +4163,7 @@ static int dna_config_check_post_op(Slapi_PBlock * pb)
             if (dna_dn_is_config(dn)) {
                 dna_load_plugin_config(pb, 0);
             }
-            if(dna_dn_is_shared_config(pb, dn) == 0){
+            if(dna_dn_is_shared_config(pb, dn)){
                 dna_load_shared_servers();
             }
         }
-- 
1.9.3

